### ISSUE TITLE
VS21019 C# version 7.2 accepts pointer to nested generic type that is rejected by VS2017

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#38105]
The following code is an error in C# 7.0 in VS 2017.  However, the same code is accepted in C# 7.0 in VS 2019.  This is a forward compatibility error (we are accepting code not part of C# 7.0 in that language version in VS 2019).

``` c#
class Program
{
    unsafe static void Main(string[] args)
    {
        G<int>.S* p = null;
    }
}

public class G<T>
{
    public struct S { int x; }
}
```

**Comment by RikkiGibson** [comment]
It seems like the following test was created to catch this scenario. It was not checking what it was probably meant to check because the usages of nested types had accessibility errors.

https://github.com/dotnet/roslyn/blob/448b9d1d4cc8bd6f6d6460349030dc45b020fc46/src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs#L2709

**Comment by agocke** [comment]
Since this regression already existed in 16.2, I think the fix can wait for 16.4
